src/server/database/clips.ts

```diff
@@ -208,17 +217,93 @@ export default class Clips {
         }
     };
 
+    /**
+     * Finds the speaker id for the input demographics
+     * @param client_id the id of the client
+     * @param gender the gender of user
+     * @param age the age of the user
+     * @param nativeLanguage the native language of the user
+     */
+    findSpeakerId = async (
+        client_id: string | undefined,
+        gender: string,
+        age: string,
+        nativeLanguage: string
+    ): Promise<string> => {
+        const [[row]] = await this.sql.query(
+            `
+                SELECT
+                    speaker_id
+                FROM
+                    clips
+                WHERE
+                    client_id = ?
+                AND
+                    sex = ?
+                AND
+                    age = ?
+                AND
+                    native_language = ?
+                AND 
+                    speaker_id is not null
+                LIMIT 1
+            `,
+            [client_id, gender, age, nativeLanguage]
+        );
+
+        // If not found, generate new speaker id
+        if (!row) {
+            const new_speaker_id = await this.generateNewSpeakerId();
+            return new_speaker_id;
+        }
+
+        const { speaker_id } = row;
+        return speaker_id;
+    };
+
+    /**
+     * Counts how many speakers there are inte the db, and
+     * returns the number + 1 as a new speaker id on a 6 character string
+     * format, the number gets padded with zeros
+     */
+    generateNewSpeakerId = async (): Promise<string> => {
+        const [[row]] = await this.sql.query(
+            `
+                SELECT 
+                    speaker_id 
+                FROM 
+                    clips 
+                ORDER BY 
+                    speaker_id desc 
+                LIMIT 1
+            `
+        );
+        const { speaker_id } = row;
+        const speaker_id_parsed = parseInt(speaker_id)
+            ? parseInt(speaker_id)
+            : 0;
+        const new_speaker_id: string = (speaker_id_parsed + 1)
+            .toString()
+            .padStart(6, '0');
+        return new_speaker_id;
+    };
+
+    /**
+     * Counts the amounts of clips with the given sentence id
+     * and updates the clips_count in the sentences table accordingly
+     * @param sentenceId the sentence to update the count for
+     */
     increaseClipCountOnSentence = async (sentenceId: string): Promise<void> => {
         await this.sql.query(
             `
             UPDATE
                 sentences
             SET
```

Comment:
this should be more accurate than the just count +1
we hade some issues that during a while this field was not updated

Reply:
I take it it's working correctly now?

Reply:
Yes should be, I trust this function more. But the original issue was that this field was not updated at all. I'm still a bit doubtful that it is the best solution to manually update a count field in the db. We should probably rely on a view or something that produces the numbers for us.

But this works, so it is for later if we want to make things prettier :)